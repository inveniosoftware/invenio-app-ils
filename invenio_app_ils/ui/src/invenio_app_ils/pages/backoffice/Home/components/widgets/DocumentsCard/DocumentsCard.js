import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Loader, Error } from '../../../../../../common/components';
import { RecordsBriefCard } from '../../../../components/statistics/RecordsBriefCard';
import { NewButton, SeeAllButton } from '../../../../components/buttons';
import {
  BackOfficeURLS,
  documentsSearchQueryUrl,
} from '../../../../../../common/urls';
import { document as documentApi } from '../../../../../../common/api';

export default class DocumentsCard extends Component {
  constructor(props) {
    super(props);
    this.fetchRequestedWithAvailableItems =
      props.fetchRequestedWithAvailableItems;
    this.seeAllUrl = documentsSearchQueryUrl;
    this.newDocumentURL = BackOfficeURLS.newDocument;
  }

  componentDidMount() {
    this.fetchRequestedWithAvailableItems();
  }

  _seeAllButton = () => {
    const handler = () =>
      this.props.history.push(
        this.seeAllUrl(
          documentApi
            .query()
            .withAvailableItems()
            .withPendingLoans()
            .qs()
        )
      );
    return <SeeAllButton fluid disabled clickHandler={() => handler()} />;
  };

  _newDocumentButton = () => {
    return (
      <NewButton
        fluid
        disabled
        clickHandler={() => this.props.history.push(this.newDocumentURL)}
      />
    );
  };

  _render_card = data => {
    return (
      <RecordsBriefCard
        title={'Documents'}
        stats={data}
        text={'requested on shelf'}
        buttonLeft={this._newDocumentButton()}
        buttonRight={this._seeAllButton()}
      />
    );
  };

  render() {
    const { data, isLoading, error } = this.props;
    return (
      <Loader isLoading={isLoading}>
        <Error error={error}>{this._render_card(data)}</Error>
      </Loader>
    );
  }
}

DocumentsCard.propTypes = {
  fetchRequestedWithAvailableItems: PropTypes.func.isRequired,
  data: PropTypes.number.isRequired,
};
