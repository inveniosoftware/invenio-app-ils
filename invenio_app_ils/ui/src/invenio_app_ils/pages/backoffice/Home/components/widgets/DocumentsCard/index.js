import { connect } from 'react-redux';
import { fetchRequestedWithAvailableItems } from './state/actions';
import DocumentsCardComponent from './DocumentsCard';
import { compose } from 'redux';
import { withRouter } from 'react-router-dom';

const mapStateToProps = state => ({
  data: state.documentsCard.data,
  error: state.documentsCard.error,
  isLoading: state.documentsCard.isLoading,
  hasError: state.documentsCard.hasError,
});

const mapDispatchToProps = dispatch => ({
  fetchRequestedWithAvailableItems: () =>
    dispatch(fetchRequestedWithAvailableItems()),
});

export const DocumentsCard = compose(
  withRouter,
  connect(
    mapStateToProps,
    mapDispatchToProps
  )
)(DocumentsCardComponent);
