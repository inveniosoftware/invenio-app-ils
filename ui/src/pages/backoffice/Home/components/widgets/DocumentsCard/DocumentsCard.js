import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Loader, Error } from '@components';
import { RecordsBriefCard } from '../../../../components/statistics/RecordsBriefCard';
import { NewButton, SeeAllButton } from '../../../../components/buttons';
import { BackOfficeRoutes } from '@routes/urls';
import { document as documentApi } from '@api';

class DataCard extends Component {
  constructor(props) {
    super(props);
    this.data = props.data;
    this.seeAllUrl = BackOfficeRoutes.documentsListWithQuery;
    this.newDocumentURL = BackOfficeRoutes.documentCreate;
  }
  newDocumentButton = () => {
    return <NewButton fluid disabled to={this.newDocumentURL} />;
  };

  seeAllButton = () => {
    const path = this.seeAllUrl(
      documentApi
        .query()
        .withAvailableItems()
        .withPendingLoans()
        .qs()
    );
    return <SeeAllButton fluid disabled to={path} />;
  };

  render() {
    return (
      <RecordsBriefCard
        title={'Documents'}
        stats={this.data}
        text={'requested on shelf'}
        buttonLeft={this.newDocumentButton()}
        buttonRight={this.seeAllButton()}
      />
    );
  }
}

export default class DocumentsCard extends Component {
  constructor(props) {
    super(props);
    this.fetchRequestedWithAvailableItems =
      props.fetchRequestedWithAvailableItems;
  }

  componentDidMount() {
    this.fetchRequestedWithAvailableItems();
  }

  render() {
    const { data, isLoading, error } = this.props;
    return (
      <Loader isLoading={isLoading}>
        <Error error={error}>
          <DataCard data={data}/>
        </Error>
      </Loader>
    );
  }
}

DocumentsCard.propTypes = {
  fetchRequestedWithAvailableItems: PropTypes.func.isRequired,
  data: PropTypes.number.isRequired,
};
