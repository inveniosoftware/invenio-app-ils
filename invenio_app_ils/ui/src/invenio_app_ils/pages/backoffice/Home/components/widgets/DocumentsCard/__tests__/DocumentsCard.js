import React from 'react';
import { shallow, mount } from 'enzyme';
import DocumentsCard from '../DocumentsCard';

jest.mock('../../../../../../../common/config');

describe('DocumentsCard tests', () => {
  let component;
  afterEach(() => {
    if (component) {
      component.unmount();
    }
  });

  it('should load the details component', () => {
    const mockedFetchDocuments = jest.fn();

    const component = shallow(
      <DocumentsCard
        data={0}
        fetchRequestedWithAvailableItems={mockedFetchDocuments}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render stats with 2 records', () => {
    const mockedFetchDocuments = jest.fn();
    const data = 2;

    component = mount(
      <DocumentsCard
        data={data}
        fetchRequestedWithAvailableItems={mockedFetchDocuments}
      />
    );

    expect(component).toMatchSnapshot();
    const stats = component.find('span').prop('data-test');
    expect(stats).toEqual(data);
  });
});
