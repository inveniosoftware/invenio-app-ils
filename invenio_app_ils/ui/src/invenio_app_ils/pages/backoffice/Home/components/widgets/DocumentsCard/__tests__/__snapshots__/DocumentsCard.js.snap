// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentsCard tests should load the details component 1`] = `
<Loader>
  <Error>
    <RecordsBriefCard
      buttonLeft={
        <NewButton
          clickHandler={[Function]}
          disabled={true}
          fluid={true}
          text="New"
        />
      }
      buttonRight={
        <SeeAllButton
          clickHandler={[Function]}
          disabled={true}
          fluid={true}
        />
      }
      stats={0}
      text="requested on shelf"
      title="Documents"
    />
  </Error>
</Loader>
`;

exports[`DocumentsCard tests should render stats with 2 records 1`] = `
<DocumentsCard
  data={2}
  fetchRequestedWithAvailableItems={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  history={[Function]}
>
  <Loader>
    <Error>
      <RecordsBriefCard
        buttonLeft={
          <NewButton
            clickHandler={[Function]}
            disabled={true}
            fluid={true}
            text="New"
          />
        }
        buttonRight={
          <SeeAllButton
            clickHandler={[Function]}
            disabled={true}
            fluid={true}
          />
        }
        stats={2}
        text="requested on shelf"
        title="Documents"
      >
        <Segment
          className="brief-card"
          raised={true}
        >
          <div
            className="ui raised segment brief-card"
          >
            <Header
              as="h3"
              textAlign="right"
            >
              <h3
                className="ui right aligned header"
              >
                Documents
              </h3>
            </Header>
            <p>
               
              <span
                data-test={2}
              >
                2
              </span>
               
              requested on shelf
            </p>
            <Grid>
              <div
                className="ui grid"
              >
                <GridRow>
                  <div
                    className="row"
                  >
                    <GridColumn
                      width={8}
                    >
                      <div
                        className="eight wide column"
                      >
                        <NewButton
                          clickHandler={[Function]}
                          disabled={true}
                          fluid={true}
                          text="New"
                        >
                          <Button
                            as="button"
                            disabled={true}
                            fluid={true}
                            icon={true}
                            labelPosition="left"
                            onClick={[Function]}
                            positive={true}
                            size="small"
                          >
                            <button
                              className="ui small fluid icon positive disabled left labeled button"
                              disabled={true}
                              onClick={[Function]}
                              tabIndex={-1}
                            >
                              <Icon
                                as="i"
                                name="plus"
                              >
                                <i
                                  aria-hidden="true"
                                  className="plus icon"
                                />
                              </Icon>
                              New
                            </button>
                          </Button>
                        </NewButton>
                      </div>
                    </GridColumn>
                    <GridColumn
                      width={8}
                    >
                      <div
                        className="eight wide column"
                      >
                        <SeeAllButton
                          clickHandler={[Function]}
                          disabled={true}
                          fluid={true}
                        >
                          <Button
                            as="button"
                            disabled={true}
                            fluid={true}
                            onClick={[Function]}
                            size="small"
                          >
                            <button
                              className="ui small fluid disabled button"
                              disabled={true}
                              onClick={[Function]}
                              tabIndex={-1}
                            >
                              See all
                            </button>
                          </Button>
                        </SeeAllButton>
                      </div>
                    </GridColumn>
                  </div>
                </GridRow>
              </div>
            </Grid>
          </div>
        </Segment>
      </RecordsBriefCard>
    </Error>
  </Loader>
</DocumentsCard>
`;
